kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -527,7 +545,34 @@ private void commitFile(String encodedPartition) {
       writers.remove(encodedPartition);
       log.debug("Removed writer for '{}'", encodedPartition);
     }
+  }
+
+  private void tagFile(String encodedPartition, String s3ObjectPath) {
+    if (!startOffsets.containsKey(encodedPartition)) {
+      log.warn("Tried to tag file with missing starting offset partition: {}. Ignoring.",
+          encodedPartition);
+    } else if (!endOffsets.containsKey(encodedPartition)) {
+      log.warn("Tried to tag file with missing ending offset partition: {}. Ignoring.",
+          encodedPartition);
+    } else if (!recordCounts.containsKey(encodedPartition)) {
+      log.warn("Tried to tag file with missing record count partition: {}. Ignoring.",
+          encodedPartition);
+    } else {
+      log.debug("Object to tag is: {}", s3ObjectPath);
+      Map<String, String> tags = new HashMap<>();
+      tags.put("startOffset", Long.toString(startOffsets.get(encodedPartition)));
+      tags.put("endOffset", Long.toString(endOffsets.get(encodedPartition)));
+      tags.put("recordCount", Long.toString(recordCounts.get(encodedPartition)));
 
-    startOffsets.remove(encodedPartition);
+      try {
+        storage.addTags(s3ObjectPath, tags);
+        log.info("Tagged file {} with starting offset {}, ending offset {}, record count {}",
+            s3ObjectPath, startOffsets, endOffsets, recordCount);
+      } catch (SdkClientException e) {
+        log.warn("Unable to tag file {}. Ignoring.", s3ObjectPath);
```

Comment:
We should add the exception here with the stacktrace. Swallowing the exception won't be very informative. 

I also wondered whether we should fail, and let it throw a retriable exception, but I think I agree with your decision to just skip tagging.

Reply:
Skip tagging works well with our internal use case.  It's something we can revisit based on user feedbacks.